### PR TITLE
clarifications and extend maps article

### PR DESCRIPTION
**Pull request by tty2**
These changes add clarification for a `nil map` behavior that could be confusing in an original text.
(nil map causes panic only on set operation, on get it is safe).

Plus extend article for `nil map` information + examples.

### CODE REVIEW

**Comment by ardan-bkennedy** [comment]
This looks good. I will try to get this updated soon.  Can you also updated the CONTRIBUTORS file please.

File: _content/tour/eng/maps/example9.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+import "fmt"
+
+type user struct {
+	name        string
+	contactInfo map[string]string
+}
+
+func main() {
+	u := user{}
```
Comment (ardan-bkennedy) [inline _content/tour/eng/maps/example9.go:17]: This needs to be   `var u user`

File: _content/tour/eng/maps.article [hunk 2]
```diff
@@ -150,6 +153,42 @@ can be run through the hash function are eligible. A good way to recognize types
...
+    var users map[string]user
+
+When create a structure that has a map as a field (technically zero value as well).
+
+    type user struct {
+        name        string
+        contactInfo map[string]string
+    }
+
+    u := user{}
```
Comment (ardan-bkennedy) [inline _content/tour/eng/maps.article:171]: Fix this as well. `var u user`

File: _content/tour/eng/maps.article [hunk 3]
```diff
@@ -27,6 +27,8 @@ block of memory underneath.
 .play maps/example5.go
 .play maps/example6.go
 .play maps/example7.go
+.play maps/example8.go
```
Comment (ardan-bkennedy) [inline _content/tour/eng/maps.article:30]: You forgot to list these two new examples above.

Comment (tty2) [reply _content/tour/eng/maps.article:32]: Thanks
Done

File: _content/tour/eng/maps/example9.go [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+	name        string
+	contactInfo map[string]string
+}
+
+func main() {
+	var u user
+
+	fmt.Printf("name value: 		`%s`\n", u.name)
+	fmt.Printf("contactInfo value:	%v\n", u.contactInfo)
+	fmt.Printf("is contactInfo nill:	%v\n", u.contactInfo == nil)
```
Comment (ardan-bkennedy) [inline _content/tour/eng/maps/example9.go:21]: nill: in the fmt call

Comment (tty2) [reply _content/tour/eng/maps/example9.go]: Feels like I mixed null and nil

**Pull request merged**